cnd/src/db/load.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{
+    db::{CreatedSwap, LiveSwap, Sqlite},
+    swap_protocols::{halight, han, LocalSwapId},
+};
+use async_trait::async_trait;
+
+/// Load data from the database.
+#[async_trait]
+pub trait Load<T>: Send + Sync + 'static {
+    async fn load(&self, swap_id: LocalSwapId) -> anyhow::Result<Option<T>>;
+}
+
+#[async_trait]
+impl Load<LiveSwap<han::LiveSwap, halight::LiveSwap>> for Sqlite {
+    async fn load(
+        &self,
+        _: LocalSwapId,
+    ) -> anyhow::Result<Option<LiveSwap<han::LiveSwap, halight::LiveSwap>>> {
+        // NOTE: When implementing this be sure to return Ok(None) if the
```

Comment:
Instead of returning an `Option` here, I would suggest the following:

- None if there is no swap with this ID
- Some if there is a swap.

The only information we learn during the communication phase is secret hash and identities, both are **not** returned from the API.
This struct can also have an enum that tells you the communication state based on what data is present in the database.
The point is, it should always return something if we have data, even if the communication is not yet finished. We are going to need this for https://github.com/comit-network/comit-rs/issues/2399

For the usecase of re-spawning watchers, where we need the additional information like secret-hash and identities, we probably want a completely different API: One that returns as all protocol watchers that need to be respawned (as a single list and not as pair of protocols).